src/detection.rs

```diff
@@ -12,16 +12,24 @@ pub(crate) fn inside_proc_macro() -> bool {
         _ => {}
     }
 
-    INIT.call_once(initialize);
-    inside_proc_macro()
+    #[cfg(feature = "is_available")]
+    {
+        proc_macro::is_available()
+    }
+
+    #[cfg(not(feature = "is_available"))]
+    {
+        INIT.call_once(initialize);
+        inside_proc_macro()
+    }
 }
 
 pub(crate) fn force_fallback() {
     WORKS.store(1, Ordering::SeqCst);
 }
 
 pub(crate) fn unforce_fallback() {
-    initialize();
+    WORKS.store(0, Ordering::SeqCst);
```

Comment:
What is the purpose of this change? This will deadlock the next call to inside_proc_macro if the INIT once has already been triggered.